### DISCUSSION TITLE
"Track quantity" not showing in API response

### DISCUSSION THREAD

**Discussion by arobbins** [Shopify/storefront-api-feedback#101] [Help]
I have a product with no options inside Shopify. The variant has both `Track quantity` and `Continue selling when out of stock` enabled.

![Screen Shot 2022-04-12 at 11 11 28 AM](https://user-images.githubusercontent.com/961947/163006873-e03d459a-9e26-4096-a51b-0827d81be09f.png)

The problem is that the response from the Storefront API does not reflect these options.

For example:

![Screen Shot 2022-04-12 at 11 15 39 AM](https://user-images.githubusercontent.com/961947/163007686-3050dff9-8e5f-483c-af79-de55cb77e80b.png)

`availableForSale`, `quantityAvailable`, and `currentlyNotInStock` return the same values regardless of what "Track quantity" and "Continue selling when out of stock" are set to.

Is there a way to detect these values in the product response?

**Comment by nboliver** [comment]
Similar issue: If you have "track quantity" on with an inventory value set, and then turn it off, the inventory value remains and there is no way to detect that inventory isn't tracked. This means that low stock UI will still show, even though unlimited stock is available. 
Possible workaround is to ensure all inventory is set to 0 before turning off "track quantity" - not ideal.

**Comment by arobbins** [comment]
Any feedback on this?

**Reply by ShahnasEk** [reply]
Hello,

Did you guys find any other workaround for the issue?

**Comment by zakariamofaddel** [comment]
This is still an issue. The Cart API `update.js` endpoint does not validate inventory (we cannot use the `change.js` endpoint), so we wanted to do it manually using the storefront API, but upon disabling the `Tack quantity` checkbox, the stock errors were still displayed.

EDIT:
For anyone facing this issue, I was able to find that the if you fetch `/variants/${variantId}.js` in the storefront, it returns a json that looks like this:
```json
{
  "id": 45766376882433,
  "title": "Default Title",
  "option1": "Default Title",
  "option2": null,
  "option3": null,
  "sku": null,
  "requires_shipping": true,
  "taxable": true,
  "featured_image": null,
  "available": true,
  "name": "The Collection Snowboard: Hydrogen",
  "public_title": null,
  "options": ["Default Title"],
  "price": 1000,
  "weight": 0,
  "compare_at_price": null,
  "inventory_management": null, // This will be "shopify" if inventory is tracked
  "barcode": null,
  "quantity_rule": { "min": 1, "max": null, "increment": 1 },
  "quantity_price_breaks": [],
  "requires_selling_plan": false,
  "selling_plan_allocations": []
}

```
And `inventory_management` has value `"shopify"` if inventory is tracked, otherwise it's `null`

**Reply by zakariamofaddel** [reply]
The solution I added does not account for the "Continue selling when out of stock" though :/

**Reply by panzerdp** [reply]
@zakariamofaddel Did you find a solution for "Continue selling when out of stock" flag in the end? Thanks.

**Reply by zakariamofaddel** [reply]
@panzerdp I ended up falling back to app proxy, where on the server I use the admin API to get the info about the `InventoryItem`, seems the only reliable way for now, unfortunately.

**Reply by panzerdp** [reply]
@zakariamofaddel Thank you. 👍

**Comment by zakariamofaddel** [comment]
Hi guys, is there any update on this or any plan to introduce a way to know the inventory policy using the Storefront API?
